File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/enterprise/info/ui/EnterpriseInfoUI.java
package com.fintechplatform.ui.enterprise.info.ui;


import android.content.Context;
import android.content.Intent;

import com.android.volley.toolbox.Volley;
import com.fintechplatform.api.enterprise.api.EnterpriseAPIModule;
import com.fintechplatform.ui.models.DataAccount;
import com.fintechplatform.api.net.NetModule;

public class EnterpriseInfoUI {

    private String hostName;
    private DataAccount configuration;

    static EnterpriseInfoUI instance;

    public EnterpriseInfoUI(String hostName, DataAccount configuration) {
        this.hostName = hostName;
        this.configuration = configuration;
    }

    EnterpriseInfoViewComponent buildLightDataViewComponent(Context context, EnterpriseInfoContract.View view) {
        return DaggerEnterpriseInfoViewComponent.builder()
                .netModule(new NetModule(Volley.newRequestQueue(context), instance.hostName))
                .enterpriseAPIModule(new EnterpriseAPIModule(instance.hostName))
                .enterpriseInfoPresenterModule(new EnterpriseInfoPresenterModule(view, instance.configuration))
                .build();
    }
    public EnterpriseInfoViewComponent createLightDataViewComponent(Context context, EnterpriseInfoContract.View view) {
        return instance.buildLightDataViewComponent(context, view);
    }

    public void start(Context context) {
        instance = this;
        Intent intent = new Intent(context, EnterpriseInfoActivity.class);
        context.startActivity(intent);
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/enterprise/info/EnterpriseInfoUI.kt
package com.fintechplatform.ui.enterprise.info

import android.content.Context
import android.content.Intent
import android.os.Bundle
import com.android.volley.toolbox.Volley
import com.fintechplatform.api.enterprise.api.EnterpriseAPIModule
import com.fintechplatform.api.net.NetModule
import com.fintechplatform.api.profile.api.IdsDocumentsAPIModule
import com.fintechplatform.ui.enterprise.info.di.DaggerEnterpriseInfoViewComponent
import com.fintechplatform.ui.enterprise.info.di.EnterpriseInfoPresenterModule
import com.fintechplatform.ui.enterprise.info.di.EnterpriseInfoViewComponent
import com.fintechplatform.ui.models.DataAccount


class EnterpriseInfoUI(private val hostName: String, private val configuration: DataAccount) {

    fun startActivity(context: Context) {
        val intent = Intent(context, EnterpriseInfoActivity::class.java)
        val args = Bundle()
        args.putString("hostname", hostName)
        args.putParcelable("dataAccount", configuration)
        intent.putExtras(args)
        context.startActivity(intent)
    }

    fun createFragment(): EnterpriseInfoFragment {
        return EnterpriseInfoFragment.newInstance(hostName, configuration)
    }

     object Builder {
         fun buildLightDataViewComponent(context: Context, view: EnterpriseInfoContract.View, hostName: String, dataAccount: DataAccount): EnterpriseInfoViewComponent {
             return DaggerEnterpriseInfoViewComponent.builder()
                     .netModule(NetModule(Volley.newRequestQueue(context), hostName))
                     .enterpriseAPIModule(EnterpriseAPIModule(hostName))
                     .idsDocumentsAPIModule(IdsDocumentsAPIModule(hostName))
                     .enterpriseInfoPresenterModule(EnterpriseInfoPresenterModule(view, dataAccount))
                     .build()
         }
     }
}